projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php

```diff
@@ -450,14 +450,14 @@ public static function user_can_edit() {
 	 * @return bool Whether the post can be viewed
 	 */
 	public static function user_can_view_post() {
-		if ( 'paid_subscribers' === self::get_newsletter_access_level() ) {
-			require_once JETPACK__PLUGIN_DIR . 'extensions/blocks/premium-content/_inc/subscription-service/include.php';
-
-			$paywall  = \Automattic\Jetpack\Extensions\Premium_Content\subscription_service();
-			$plan_ids = self::get_all_plans_id_jetpack_recurring_payments();
-			return $paywall->visitor_can_view_content( $plan_ids );
+		$newsletter_access_level = self::get_newsletter_access_level();
+		if ( 'everybody' === $newsletter_access_level ) {
```

Comment:
I wonder if this test should directly be in `visitor_can_view_content` to prevent logic to be scattered everywhere...

Also, I think we are missing the use case where `self::get_newsletter_access_level()` returns null, or '' ( I think it can happen, but I could be wrong) - for example for posts created before enabling the newsletter plugin, `get_newsletter_access_level` does not take that into account...

Reply:
> Also, I think we are missing the use case where self::get_newsletter_access_level() returns null, or '' ( I think it can happen, but I could be wrong) - for example for posts created before enabling the newsletter plugin, get_newsletter_access_level does not take that into account...

Great point!

> I wonder if this test should directly be in visitor_can_view_content to prevent logic to be scattered everywhere...

I have no objections

Reply:
Fixed in [e859057](https://github.com/Automattic/jetpack/pull/27212/commits/e859057e4a99ab6468025cdeaebf03b85825627b)